lib/reporting/cloudwatch_client.rb

```diff
@@ -117,8 +118,12 @@ def fetch(query:, from: nil, to: nil, time_slices: nil)
           log(:debug, "thread done thread_idx=#{thread_idx}")
 
           nil
+        rescue => err
+          errors << err
+          queue.clear
+          in_progress.clear
         end.tap do |thread|
-          thread.abort_on_exception = true
+          thread.report_on_exception = false
```

Comment:
`abort_on_exception` meant the main thread would throw an error. useful for stopping everything when running a script locally, but detrimental in a backgroiund job

`report_on_exception` creates a bunch of console barf, so turning this off cleaned up specs